### ISSUE TITLE
Mattermost integration

### ISSUE THREAD

**Issue by ctron** [eclipse/packagedrone#25]
Implement a integration for sending events from Package Drone to Mattermost [1].

Events like artifact added or remove could be sent.

[1] http://www.mattermost.org/

**Comment by kdoteu** [comment]
I think a general API would be better. So people (like me and me company) can integrate this in their chat systems for example Slack or Hipchat ?

**Comment by ctron** [comment]
Good point. I know that Mattermost's push events are compatible with Slack. Don't know about HipChat though.

The question is, how can the Package Drone side be made compatible? Because AFAIK, web-hooks would mean that Package Drone does define the format and all other would need to implement that. Which I guess won't happen for some time.

**Comment by doggy-dev** [comment]
Maybe a configuration file uploaded to a channel (or every channel) could define the hook parameters. i.e.
```
{
  host:"http://chat.server.com",
  params:[
    p1:"${messageBody}",
    p2:"${channel}"
  ]
}
```
